hanadb_exporter/exporters/prometheus_exporter.py

```diff
@@ -31,48 +36,71 @@ class SapHanaCollector(object):
     """
 
-    def __init__(self, conector):
-        super(SapHanaCollector, self).__init__()
+    def __init__(self, connector):
         self._logger = logging.getLogger(__name__)
-        self._hdb_connector = conector
+        self._hdb_connector = connector
 
-    def _execute(self, metric):
+
+    def _format_query_result(self, query_result):
         """
-        Create metric object
+        Format query results to match column names with their values for each row
+        Returns nested list, containing tuples (column_name, value)
 
         Args:
-            metric (dict): query, info, type structure dictionary
+            query_result (obj): QueryResult object
         """
-        try:
-            value = self._hdb_connector.query(metric['query'])
-
-            if metric['type'] == core.GaugeMetricFamily:
-                metric_obj = self._manage_gauge(metric, value)
-            else:
-                raise NotImplementedError('{} type not implemented'.format(metric['type']))
-
-            return metric_obj
-        except KeyError as err:
-            raise MalformedMetric(err)
-
-    def _manage_gauge(self, metric, value):
+        query_columns = []
+        formatted_query_result = []
+        query_columns = [meta[0] for meta in query_result.metadata]
+        for record in query_result.records:
+            formatted_query_result.append(list(zip(query_columns, record)))
+        # TODO manage formatted_query_result with a class, a named tuple or a dictionary instead of a tuple
+        return formatted_query_result
+
+    def _manage_gauge(self, metric, formatted_query_result):
         """
         Manage Gauge type metric
+
+        Args:
+            metric (dict): a dictionary containing information about the metric
+            formatted_query_result (nested list): query formated by _format_query_result method
         """
-        # Label not set
-        metric_obj = core.GaugeMetricFamily(*metric['info'])
-        if not metric['info'][3]:
-            metric_obj.add_metric([], str(value[0][0]))
-        else:
-            for label_item in value:
-                self._logger.info('%s: %s' % (label_item[0], label_item[1]))
-                metric_obj.add_metric([label_item[0]], float(label_item[1]))
+        metric_obj = core.GaugeMetricFamily(metric['name'],
+            metric['description'], None, metric['labels'], metric['unit'])
+        for row in formatted_query_result:
+            labels = []
+            value = None
+            for cell in row:
+                # each cell is a tuple (column_name, value)
+                if cell[0] in metric['labels']:
+                    labels.append(cell[1])
+                if metric['value'] == '':
+                    raise ValueError('No value specified in metrics.json for {}'
+                        .format(metric['name']))
+                elif cell[0] == metric['value']:
+                    value = cell[1]
+            if value is None:
```

Comment:
I'm not sure if this is a good solution. With this approach, retrieving a unique invalid value would make fail the whole reporting. Have you tested this scenario?

Reply:
yes it will, since the config file in that case would be invalid, what do you propose as an alternative ?

Reply:
mmm, could we just skip that value? Or prometheus requests all the metrics always?

Reply:
@arbulu89 what do you mean by skipping the value? like the result would be Null for example ?